test/frame.jl

```diff
@@ -135,6 +135,15 @@
         @test count_residues(Topology(frame)) == 1
     end
 
+    @testset "Frame indexing" begin
+        frame = Frame()
+        add_atom!(frame, Atom("Zn"), [0.0, 0.0, 0.0])
+        add_atom!(frame, Atom("Fe"), [1.0, 2.0, 3.0])
+
+        @test name(frame[0]) == "Zn"
+        @test name(frame[1]) == "Fe"
```

Comment:
Could you also add tests checking that one can write to the frame/topology using indexing?

Reply:
First of all, sorry for the long delay, I had a lot of work-related things to do in the last weeks, and very little time for open source projects.

---

> Anyway, I tried the equivalent of that with this PR, i.e: [...] and this ran fine.

This issue only manifest itself randomly, depending on the state of the OS memory allocator.

> Even if this can fail, I think it would be okay since accessing a reference to a data after it has been changed anyway would have undefined behavior. If I understand about what you said correctly, shouldn't this also be an issue in the C++ interface?

Yes, it is also an issue with the C and C++ interfaces. I'm trying to make it less of a problem on the Julia side, since Julia users are not used to think in these terms =). 

--- 

Overall, I think I'm fine with merging this PR, with some documentation for these functions indicating that one should not old a reference to an Atom obtained this way.